File: ntra/event/task.py
import frappe
def after_save(doc, method):
    p_task_subject = frappe.db.get_value("Task", doc.parent_task, "subject")
    p_goal_name = frappe.db.get_value("Goal", {"goal_name":p_task_subject, "is_group": 1})
    frappe.db.set_value("Goal", {"goal_name": doc.subject, "parent_goal": p_goal_name}, "custom_weight_progress", doc.progress)
    frappe.db.set_value("Goal", doc.custom_objective, "custom_weight_progress", doc.progress)

@frappe.whitelist()
def update_complete(objective):
    frappe.db.set_value("Task", objective, "status", "Completed")
    frappe.db.set_value("Task", objective, "progress", 100)
    frappe.db.commit()
    return True

File: ntra/public/js/task.js
frappe.ui.form.on("Task", {
	refresh(frm) {   
        if(frm.doc.custom_is_edit && frm.doc.status != "Completed"){
            frm.add_custom_button('Edit Objective', () => {
                let dialog = new frappe.ui.Dialog({
                    title: 'Custom Dialog',
                    fields: [
                        {
                            label: 'Goal Name',
                            fieldname: 'goal_name',
                            fieldtype: 'Data',
                            reqd: 1
                        },
                        {
                            label: 'Description',
                            fieldname: 'description',
                            fieldtype: 'Text',
                            reqd: 1
                        }
                    ],
                    primary_action_label: 'Submit',
                    primary_action(values) {
                        // Perform an action with the dialog input
                        frappe.call({
                            method: "ntra.event.objective.edit_objective",
                            args: {
                                goal_name: values.goal_name,
                                description: values.description,
                                objective: frm.doc.custom_objective
                            },
                            callback: function(r){
                                if(r.message){
                                    frappe.msgprint(__("Objective is edited Successfully"))
                                    frappe.call({
                                        method: "ntra.event.task.update_complete",
                                        args: {
                                            objective: frm.doc.name
                                        },
                                        callback: function(r){
                                            if(r.message){
                                                frm.refresh()
                                            }
                                        }
                                    })
                                }
                            }
                        })
                        dialog.hide();
                    }
                });
                dialog.show()
            }).css({color: 'red'});
        }
    }
})